src/interface/metadata.cpp

```diff
@@ -98,6 +98,71 @@ MetadataFlag Metadata::GetUserFlag(const std::string &flagname) {
 }
 
 namespace parthenon {
+Metadata::Metadata(const std::vector<MetadataFlag> &bits, const std::vector<int> &shape,
+                   const std::vector<std::string> &component_labels,
+                   const std::string &associated)
```

Comment:
Was moving the constructor into the implementation file what fixed compile time issues?

Reply:
yes, at least to a large extent. The compile time still isn't great on GPU.